Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Indexer/NodeIndexer.php

```diff
@@ -398,6 +398,35 @@ public function flush() {
 	}
 
 	/**
+	 * @param string $bulkRequest
+	 * @param string $errors
+	 */
+	protected function logIndexingErrors($bulkRequest, $errors) {
+		if (!file_exists(FLOW_PATH_DATA . 'Logs/ElasticSearch')) {
+			mkdir(FLOW_PATH_DATA . 'Logs/ElasticSearch');
+		}
+		if (file_exists(FLOW_PATH_DATA . 'Logs/ElasticSearch') && is_dir(FLOW_PATH_DATA . 'Logs/ElasticSearch') && is_writable(FLOW_PATH_DATA . 'Logs/ElasticSearch')) {
+			$referenceCode = date('YmdHis', $_SERVER['REQUEST_TIME']) . substr(md5(rand()), 0, 6);
+			$dumpPathAndFilename = FLOW_PATH_DATA . 'Logs/ElasticSearch/' . $referenceCode . '.txt';
+			file_put_contents($dumpPathAndFilename, $this->renderIndexingErrors($bulkRequest, $errors));
```

Comment:
I highly discourage this custom approach. Instead use the proper logging API, see an example here https://github.com/mocdk/MOC.Varnish/commit/06f8b8f49ef34bc28a0c74fab677d6d80a024e81